### ISSUE TITLE
Font weights are too light

### ISSUE THREAD

**Issue by m4rc1e** [canonical/Ubuntu-fonts#99]
Somewhat related to #82.

The font weights are off. The Regular is now much lighter than the family with serve on Google Fonts. Same can be said for Medium, Bold etc.

<img width="462" alt="Screenshot 2023-07-05 at 13 49 31" src="https://github.com/canonical/Ubuntu-fonts/assets/7525512/427f78c2-bc2b-4634-9c14-1fbd01484310">

*old=bottom, new=top*

I haven't looked at this family in 8+ years but I think the original static fonts were not Multiple Master friendly.

**Comment by djrrb** [comment]
Lightening the Regular weight and rearranging the others accordingly was an intentional change requested by Canonical to adapt the font to current UI tastes / screen resolutions. The original weight is now closer to `wght=473`. 

I’m not sure how this and other [significant changes](https://github.com/djrrb/google-fonts/blob/main/ufl/ubuntu/FONTLOG.txt#L29) should be handled in respect to the upgrade to Google Fonts, but these are the weights that Canonical plans to use for Ubuntu moving forward.

**Comment by m4rc1e** [comment]
Thank you for the explanation. 

I worry I'm going to open issues for regressions which have been approved. Do you have a doc anywhere of known changes?

**Comment by djrrb** [comment]
I’m not sure I have much aside from this summary in the log: https://github.com/djrrb/google-fonts/blob/main/ufl/ubuntu/FONTLOG.txt#L29-L43

Were you looking for something more detailed? Also happy to quickly talk through the changes on a call if that’s quicker.

**Comment by fitojb** [comment]
Even Canonical web designers have found (tacitly, they won’t admit it verbally, but their actions show otherwise) the new weight distribution to be too light; my evidence for asserting this is that the new Vanilla Design System version 4.0 [uses a weight of 180](https://vanillaframework.io/docs/base/typography#:~:text=Display%20headings-,180,-H2%2C%20large%20screens) instead of a more intuitive 100 for H2 headings. This kind of weight finessing isn’t available to desktop-application libraries like GTK and Qt, of course, so it’s become a pain point when used as a UI font. Ubuntu OS users need better defaults, as I’ve expressed [here](https://discourse.ubuntu.com/t/ubuntu-font-update-available-for-testing/34676/20?u=fito).

**Comment by djrrb** [comment]
@fitojb I don’t think it’s my place to comment on the old vs new weight arrangement, but I hope it’s at least a little helpful to mention the following:

> the new Vanilla Design System version 4.0 [uses a weight of 180](https://vanillaframework.io/docs/base/typography#:~:text=Display%20headings-,180,-H2%2C%20large%20screens) instead of a more intuitive 100 for H2 headings.

We just added SemiLight (200) weight, very close to 180: https://github.com/canonical/Ubuntu-fonts/issues/97#issuecomment-1625538124

I placed the SemiLight exactly at the midpoint between Thin and Light, which I think is a sensible choice for the type family. But other placements are possible if the design team requests it.

> This kind of weight finessing isn’t available to desktop-application libraries like GTK and Qt,

Maybe it should be? 

> so it’s become a pain point when used as a UI font.

I get it. FWIW it is possible to generate new static fonts for off-instance locations directly from the variable fonts. [fontTools varLib.instancer](https://fonttools.readthedocs.io/en/latest/varLib/instancer.html) provides a command line interface, and [FontGauntlet](http://fontgauntlet.com) offers a “Generate Static Font File” button.

As a quick demo, here are instancer-generated fonts where I also changed the menu names to “Ubuntu 180” to distinguish them: [Ubuntu180.zip](https://github.com/canonical/Ubuntu-fonts/files/11994059/Ubuntu180.zip)

This is probably worth adding to the readme, so folks know it’s an option.

**Comment by barmadrid** [comment]
> As a quick demo, here are instancer-generated fonts where I also changed the menu names to “Ubuntu 180” to distinguish them: [Ubuntu180.zip](https://github.com/canonical/Ubuntu-fonts/files/11994059/Ubuntu180.zip)
> 

This is really awesome. Can you please let me know the command used for Instancer to generate a specific weight and change font menu name? Couldn't figure it our from the docs :)

**Comment by djrrb** [comment]
@barmadrid 

Here’s an example of how I used [fontTools varLib.instancer](https://fonttools.readthedocs.io/en/latest/varLib/instancer.html), now included in this repo’s [readme](https://github.com/canonical/Ubuntu-fonts/blob/main/README.md?plain=1#L46).

```fonttools varLib.instancer Ubuntu[wdth,wght].ttf wdth=100 wght=473```

I used the following process to change the menu names:

1. `ttx Ubuntu[wdth,wght].ttf` converts the font to XML
2. Open resulting `Ubuntu[wdth,wght].ttx` in a text editor
3. Run a find/replace in the `<name>` table
4. `ttx Ubuntu[wdth,wght].ttx` to convert it back to a TTF

I think you could also use [pyftfeatfreeze](https://github.com/twardoch/fonttools-opentype-feature-freezer) to accomplish the same task.

```pyftfeatfreeze -R 'Ubuntu/MyCustomFont' Ubuntu[wdth,wght].ttf MyCustomFont[wdth,wght].ttf```

Hope that helps!

**Comment by lyubomir-popov** [comment]
Hi just seeing this discussion, I feel a bit of explanation is necessary when referencing Vanilla 4.0's font choices. 
We make use of the full range of weights in our design system, reserving Thin (100) for Display headings only (font-size 80px) (for example, see the canonical.com homepage). 
For smaller headings, including the afore mentioned h2, we aim to achieve uniform (perceived, not mathematical) stem thickness for Display, h2, h4 and p styles, hence our more surgical use of values like 180.

**Comment by up-to-you** [comment]
Was able to successfully convert variable fonts to wght=475 and to alter style names (R,B,I,BI). But it was quite time consuming task. I assume that everyone will be happy if you will just add Book/Text weight (473/475 up to your perfectional discretion). A lot of people on Linux are used to bolder MacOS font rendering, such that Ubuntu is one of the few (also Rubik is popular choice, but it's boring) fonts which has bolder (473/475) Regular style.

**Comment by mwskwong** [comment]
I agree. I would say setting +50 to all font weights across the board feels better match to what other fonts look like.
